src/config.rs

```diff
@@ -0,0 +1,153 @@
+pub mod storage;
+pub mod settings;
+
+use std::collections::HashMap;
+use serde_derive::Deserialize;
+use serde_json::Value;
+use wildmatch::WildMatch;
+use crate::config::settings::{Setting, SettingInfo};
+
+const SETTINGS_JSON: &str = include_str!("./config/settings.json");
+
+#[derive(Debug, Deserialize)]
+struct JsonEntry {
+    key: String,
+    #[serde(rename = "type")]
+    entry_type: String,
+    default: String,
+    description: String,
+}
+
+trait StorageAdapter {
+    /// Retrieves a setting from the storage
+    fn get_setting(&self, key: &str) -> Option<Setting>;
+    /// Stores a given setting to the storage
+    fn set_setting(&mut self, key: &str, value: Setting);
+    /// Retrieves all the settings in the storage
+    fn get_all_settings(&self) -> HashMap<String, Setting>;
+}
+
+/// Configuration store is the place where the gosub engine can find all configurable options
+pub struct ConfigStore {
+    /// A hashmap of all settings so we can search o(1) time
+    settings: HashMap<String, Setting>,
+    /// A hashmap of all setting descriptions
+    settings_info: HashMap<String, SettingInfo>,
+    /// Keys of all settings so we can iterate keys easily
+    setting_keys: Vec<String>,
```

Comment:
I'm not sure if we need this, but if we need to iterate all keys, we need to store them in a vec as well.